### PR TITLE
kubelet: add setting for configuring serverTLSBootstrap

### LINKED ISSUE #1467: add setting for kubelet server TLS bootstrap
**What I'd like:**
Today we unconditionally set `serverTLSBootstrap: true` in the kubelet config. This causes issues with Cluster API provisioning, at least with the CAPA provider, since the CSRs that the kubelet submits are not automatically signed.

If we exposed a setting for this, we could turn off this behavior in cases where we knew it would not cause problems.

**Any alternatives you've considered:**
Continue setting the value to true and require users to install a solution like [kubelet-rubber-stamp](https://github.com/kontena/kubelet-rubber-stamp) to sign the resulting CSRs.

### PR DESCRIPTION
**Pull request by etungsten**
**Issue number:**
Fixes https://github.com/bottlerocket-os/bottlerocket/issues/1467

**Description of changes:**
```
Author: Erikson Tung <etung@amazon.com>
Date:   Tue Apr 13 13:58:05 2021 -0700

    migrations: add migration for adding `kubernetes.server-tls-bootstrap`
    
    Adds a migration for the new `kubernetes.server-tls-bootstrap` setting.

```

```
Author: Erikson Tung <etung@amazon.com>
Date:   Tue Apr 13 13:40:59 2021 -0700

    k8s: add setting for configuring serverTLSBootstrap
    
    Adds a new setting `kubernetes.server-tls-bootstrap` for configuring
    whether to enable server certificate bootstrap for the kubelet.

```

**Testing done:**
Built x86 aws-k8s-1.19 image and launched instance with said image. Was able to toggle `kubernetes.server-tls-bootstrap` setting and see kubelet-config update accordingly, kubelet was able to restart successfully after each settings change. Node is still ready and can still run pods.

Launching the instance with userdata that sets `kubernetes.server-tls-bootstrap` to `false` also works. The node comes up fine and can run pods.

When the instance is launched with `kubernetes.server-tls-bootstrap` to `false`. `kubectl get csr` does not show any CSRs from the launched node. When I toggle the setting to true, `kubectl get csr` shows the node requesting a new CSR.

When the instance is launched with `kubernetes.server-tls-bootstrap` to `true`. `kubectl get csr` does show a CSR from the launched node.

Tested migration by upgrading from the release image to the image with the setting and saw that I was able to toggle the new setting as expected. Downgraded back to the release version and the setting no longer exists as expected and `serverTLSBootstrap` is set to true by default as expected.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/api/migration/migrations/v1.1.0/kubelet-server-tls-bootstrap/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new settings for configuring kubelet, `kubernetes.server-tls-bootstrap`
```
Comment (etungsten) [inline sources/api/migration/migrations/v1.1.0/kubelet-server-tls-bootstrap/src/main.rs]: typo: 
```suggestion
/// We added a new setting for configuring kubelet, `kubernetes.server-tls-bootstrap`
```

File: README.md [hunk 2]
```diff
@@ -309,6 +309,7 @@ The following settings are optional and allow you to further configure your clus
 * `settings.kubernetes.cluster-domain`: The DNS domain for this cluster, allowing all Kubernetes-run containers to search this domain before the host's search domains.  Defaults to `cluster.local`.
 * `settings.kubernetes.standalone-mode`: Whether to run the kubelet in standalone mode, without connecting to an API server.  Defaults to `false`.
 * `settings.kubernetes.authentication-mode`: Which authentication method the kubelet should use to connect to the API server, and for incoming requests.  Defaults to `aws` for AWS variants, and `tls` for other variants.
+* `settings.kubernetes.server-tls-bootstrap`: Whether to enable server certificate bootstrap.  When enabled, the kubelet will request a certificate from the certificates.k8s.io API.  This requires an approver to approve the certificate signing requests (CSR).  Defaults to `true`.
```
Comment (zmrow) [inline README.md]: Nit: might be clearer to say "Enables or disables server certificate bootstrap.  When enabled, ..."

**Comment by etungsten** [comment]
Push above addresses @zmrow 's comment. Fixes wording in README.

**Comment by etungsten** [comment]
Updated testing description to show that when `server-tls-bootstrap` is set to false, kubelet does not request any CSR.

**Comment by etungsten** [comment]
Push above fixes a typo.

**Review by bcressey — APPROVED** [review]

**Review by zmrow — APPROVED** [review]
:t-rex:

**Review by webern — APPROVED** [review]

**Comment by etungsten** [comment]
Push above rebases onto develop and fixes conflicts.

**Pull request merged**